### ISSUE TITLE
[java] DanglingJavadoc FP in module-info files

### ISSUE THREAD

**Issue by jorsol** [pmd/pmd#6058]
**Affects PMD Version:** 
7.17.0

**Rule:**
DanglingJavadoc

Please provide the rule name and a link to the rule documentation:
https://docs.pmd-code.org/pmd-doc-7.17.0/pmd_rules_java_documentation.html#danglingjavadoc

**Description:**

**Code Sample demonstrating the issue:**

```java
/**
 * My Modules Documentation.
 *
 * @since 1.0
 */
module my.module.name {
  exports my.package.name;
  requires org.jspecify;
}
```

**Expected outcome:**

PMD reports a violation at line 1, but that's wrong. That's a false positive. Modules can also have JavaDoc.

**Running PMD through:** *Maven*
